processing/src/main/java/io/druid/segment/filter/AndFilter.java

```diff
@@ -181,6 +170,9 @@ public String toString()
 
   private ValueMatcher makeMatcher(final ValueMatcher[] baseMatchers)
   {
+    if (baseMatchers.length == 0) {
```

Comment:
Please add a check in the constructor and change this to check too like in OrFilter.
Also, please remove [these lines](https://github.com/metamx/druid/blob/5633e9771b2c5620905745c3b73656d992fcc9e9/processing/src/main/java/io/druid/segment/filter/AndFilter.java#L84-L86).